.github/actions/update-best-practice-table/main.py

```diff
@@ -0,0 +1,159 @@
+            if rst_match and previous_line.strip():
+                current_heading = previous_line.strip()
+                previous_line = line
+                continue
+
+        if file_path.suffix == '.md':
+            if re.match(r'^```{admonition} Best practice', line):
+                inside_admonition = True
+                admonition_lines.clear()
+                previous_line = line
+                continue
+        else:  # .rst
+            if re.match(r'^\.\. admonition:: Best practice', line):
+                inside_admonition = True
+                admonition_lines.clear()
+                previous_line = line
+                continue
+
+        if inside_admonition:
+            if file_path.suffix == '.md':
+                at_end = line.strip() == '```'
+            else:  # .rst
+                at_end = previous_line.strip() == '' and len(line) > 0 and line[0] != ' '
+            if at_end:
+                results.append((current_heading, current_ref, '\n'.join(admonition_lines)))
+                inside_admonition = False
+            else:
+                admonition_lines.append(line)
+
+        if file_path.suffix == '.md':
+            ref_match = re.match(r'\((.+?)\)=', line)
+        else:  # .rst
+            ref_match = re.match(r'.. _(.+?):', line)
+        if ref_match:
+            current_ref = ref_match.group(1)
+            continue
+
+        previous_line = line
+
+    return results
+
+
+def make_ops_ref(heading: str, raw: str):
+    """Turn the raw reference into a link."""
+    return f'[{heading}](#{raw})'
+
+
+def make_charmcraft_ref(heading: str, raw: str):
+    """Turn the charmcraft reference into an external intersphinx link."""
+    return f'{{external+charmcraft:ref}}`{heading.replace("`", "")} <{raw}>`'
+
+
+def main():
+    """Extract the best practices, update the doc, and open a PR."""
+    parser = argparse.ArgumentParser(
+        description='Open a pull request to update the list of best practices. '
+        'Run in a clone of the operator repository.'
+    )
+    parser.add_argument(
+        '--path-to-charmcraft',
+        type=pathlib.Path,
+        default=pathlib.Path(__file__).parent.parent.parent.parent / 'charmcraft',
+        help='Path to a clone of canonical/charmcraft',
+    )
+    args = parser.parse_args()
+    path_to_ops = pathlib.Path(__file__).parent.parent.parent.parent
+    for directory, base_url, make_ref in (
+        (path_to_ops / 'docs', 'https://documentation.ubuntu.com/ops/latest/', make_ops_ref),
+        (
+            args.path_to_charmcraft / 'docs',
+            'https://documentation.ubuntu.com/charmcraft/stable/',
+            make_charmcraft_ref,
+        ),
+    ):
+        for file_path in directory.rglob('*'):
+            if file_path.suffix not in ('.md', '.rst'):
+                continue
+            if file_path.name == 'best-practices-generated':
+                continue
```

Comment:
Is this meant to be a full file name?
I imagine you meant to write:

```suggestion
            if file_path.stem == 'best-practices-generated':
                continue
```